api/cluster/resource/templater.go

```diff
@@ -187,21 +188,25 @@ func (t *InferenceServiceTemplater) CreateInferenceServiceSpec(modelService *mod
 func (t *InferenceServiceTemplater) createPredictorSpec(modelService *models.Service) kservev1beta1.PredictorSpec {
 	envVars := modelService.EnvVars
 
-	// Set cpu limit and memory limit to be 2x of the requests
-	cpuLimit := modelService.ResourceRequest.CPURequest.DeepCopy()
-	cpuLimit.Add(modelService.ResourceRequest.CPURequest)
-	memoryLimit := modelService.ResourceRequest.MemoryRequest.DeepCopy()
-	memoryLimit.Add(modelService.ResourceRequest.MemoryRequest)
+	// Set resource limits to request * userContainerCPULimitRequestFactor or UserContainerMemoryLimitRequestFactor
+	limits := map[corev1.ResourceName]resource.Quantity{}
+	if t.deploymentConfig.UserContainerCPULimitRequestFactor != 0 {
```

Comment:
possible null?

Reply:
No it wouldn't be null:

https://github.com/caraml-dev/merlin/blob/main/api/cluster/resource/templater.go#L115